Added main features (#74)
miamtech/mealz-documentation: docs/web_ssr/main-features/recipe-card.md

---
sidebar_position: 2
---

# Recipe card (SSR)

## Overview

The recipe card is the base of Mealz' whole experience. This card fetches and shows a recipe based on the items that are displayed around it (see [here](../about/features#how-the-recipes-displayed-in-the-cards-are-chosen) to see how the process of fetching the best recipe works).

> The primary CTA on the card is the "Baskets icon" button. When clicked, Mealz' drawer opens and shows the recipes' details component. Clicking on the picture or title has the same effect.

Reviewer: ```suggestion
The primary CTA on the card is the "Baskets icon" button. When clicked, Mealz's drawer opens and shows the recipe's details. Clicking on the picture or title has the same effect.
```

> Finally, by clicking on the heart icon the user can add the recipe to Mealz' favorite recipes list. This icon only appears if the user is connected

Reviewer: ```suggestion
Finally, by clicking on the heart icon the user can add the recipe to Mealz's favorite recipes list. This icon only appears if the user is connected.
```

![Recipe card](https://storage.googleapis.com/assets.miam.tech/kmm_documentation/web/examples/recipeCard.png "Recipe card")

> > You need to have loaded the **token** to display recipes and the library needs a valid **store** to add products to cart

Reviewer: ```suggestion
> You need to have loaded the **token** to display recipes and the library needs a valid **store** to add products to the cart
```

- Parameters :

  - `store_id: string`:
  **_(Mandatory)_** Your store ID

- `recipe_id: string`:
  **_(Optional)_** load a recipe from its id

- `surrounding_products_ids: string[]`:
  **_(Mandatory if no `recipe_id` provided)_** an array of productIds around the place where you want to put the recipe card, the goal is to find a recipe matching the products
  _eg: if the recipe is surrounded by milk, we want to show a recipe that needs milk_

> 

Reviewer: ```suggestion
```

  - `pricebook_key: string = 'DEFAULT'`:
  **_(Optional)_** the pricebook key is needed to retrieve the recipe price

  - `profiling: boolean = true`:
  **_(Optional)_** set to false if user refused profiling

  - `display_variant: number = 1`:

>   **_(Optional)_** Select the variant for the display of the card. Default is 1, available values are 1, 2 and 3. See below for examples.

Reviewer: ```suggestion
  **_(Optional)_** Select the variant for the display of the card. Default is 1, available values are 1, 2 and 3 (see below for examples)
```

pour rester cohérent et éviter les . à la fin des descriptions

  - `orientation: 'vertical' | 'horizontal' = 'vertical'`:

>   **_(Optional)_** Select the orientation for the display of the card. See below for examples.

Reviewer: ```suggestion
  **_(Optional)_** Select the orientation for the display of the card (see below for examples)
```

  - `current_products_ids: string[]`:
  **_(Optional)_** takes an array of product ids with a high priority on the suggestion

  - `serves: number = 4`

>   **_(Optional)_** serves parameter indicate the number of people the meal will serve.

Reviewer: ```suggestion
  **_(Optional)_** serves parameter indicate the number of people the meal will serve
```

  **_(Mandatory)_** Your store ID

  - `recipe_id: string`:
  **_(Optional)_** load a recipe from its id

  - `surrounding_products_ids: string[]`:
  **_(Mandatory if no `recipe_id` provided)_** an array of productIds around the place where you want to put the recipe card, the goal is to find a recipe matching the products
  _eg: if the recipe is surrounded by milk, we want to show a recipe that needs milk_

  - `pricebook_key: string = 'DEFAULT'`:
  **_(Optional)_** the pricebook key is needed to retrieve the recipe price

  - `profiling: boolean = true`:
  **_(Optional)_** set to false if user refused profiling

  - `display_variant: number = 1`:
  **_(Optional)_** Select the variant for the display of the card. Default is 1, available values are 1, 2 and 3 (see below for examples)

  - `orientation: 'vertical' | 'horizontal' = 'vertical'`:
  **_(Optional)_** Select the orientation for the display of the card (see below for examples)

  - `current_products_ids: string[]`:
  **_(Optional)_** takes an array of product ids with a high priority on the suggestion

  - `serves: number = 4`
  **_(Optional)_** serves parameter indicate the number of people the meal will serve

> > As mentioned earlier, while you have the possibility to display a random recipe, we heavily recommend you to pass the products ids, because while a user would be interested by a chocolate cake recipe between chocolate bars, they could find it weird to have for instance a burger recipe between 2 bottles of milk.
> > On the same note, displaying a fixed recipe can be tempting if you want to have control on the content appearing on your website, but it also means that the users will always see the same recipes at the same places and may not be interested, while our algorithm has some random components that makes the content vary between sessions, giving the user more inspiration.

Reviewer 2: ```suggestion
> Displaying a fixed recipe can be tempting if you want to have control on the content appearing on your website, but it also means that the users will always see the same recipes at the same places and may not stay interested, while our algorithm has some random elements to it that makes the content vary between sessions, giving the user more inspiration.
```
on a retiré l'Input random donc on vire le 1er paragraphe

### Example :

:::warning

> Do not forget [mandatory HTTP headers](./pre-rendered-components#http-request-headers)

Reviewer 2: ```suggestion
Do not forget the [mandatory HTTP headers](./pre-rendered-components#http-request-headers)
```
